Percussion panel - implement keyboard navigation for toolbar and pad grid

mathesoncalum opened the pull request:
This PR lays the groundwork for keyboard navigation in the percussion panel. A handful of to-dos remain for future percussion panel PRs:

- [x] Layout editing should be possible using the keyboard only
- [ ] Escape button handling
- [ ] Some subtle aspects of navigating up/down between rows are not to-spec yet.
- [ ] Focus should move to the first pad in a new row when the "add row" button is triggered via keyboard.

Eism commented on src/notation/qml/MuseScore/NotationScene/internal/PercussionPanelToolBar.qml:
```diff
@@ -57,7 +60,8 @@ Item {
         x: prv.centerX - (centralButtonsRow.width / 2)
 
-        // We only want to round the outer corners of the buttons...
+        // Workaround - you can't change the radius of specific corners in this version of Qt Quick (and we
```
Can we use RoundedRectangle.qml here?

Eism commented on src/notation/qml/MuseScore/NotationScene/internal/PercussionPanelToolBar.qml:
```diff
@@ -132,14 +176,21 @@ Item {
+        id: rightSideNavPanel
+        name: "PercussionPanelToolBarRightSide"
+        section: root.navigationSection
+        order: root.navigationOrderStart + 2
```
The navigation ought to be sequential, so it should be `order: centralNavPanel.order + 1`

Eism commented on src/notation/qml/MuseScore/NotationScene/internal/PercussionPanelPad.qml:
```diff
@@ -41,6 +41,13 @@ DropArea {
 
+    property int navigationRow: -1
+    property int navigationColumn: -1
+    property alias padNavigationCtrl: padNavCtrl
```
It's better to provide a panel property, it looks more logical for this file

mathesoncalum replied:
I decided on this approach because there are actually two panels (one for the pads and one for the footers) which have the same navigation row/column.

Eism replied:
two options: give two panels or remove Ctrl from the name (like for example for FlatButton)
I don't like Ctrl because it clearly shows that we are giving the private part out

Eism commented on src/notation/qml/MuseScore/NotationScene/internal/PercussionPanelToolBar.qml:
```diff
@@ -110,15 +116,53 @@ Item {
+        id: writeFocusBorder
+
+        anchors {
+            fill: null
```
you set it lower for all sides, is it unnecessary here?

shoogle commented:
I've tested the latest build. It works well, but I found the following issues:

1. The Percussion panel comes after the bottom bar (Workspace, etc.) in the Tab & F6 order.
    - Instead, the panel should go _before_ the bottom bar in the Tab & F6 order.
2. Pressing the "Write" button immediately puts you in note input mode.
    - Instead, it should remain in normal mode until the user presses one of the pads.
3. After exchanging the position of two pads via keyboard, all pads gain a blue highlight and focus leaves the panel.
    - Instead, all pads should _lose_ the highlight, and focus should remain on the destination pad.
4. The "Reset layout" option does nothing.
5. Ideally, the screen reader text for pads & footers should be like this:

Image | ![image](https://github.com/user-attachments/assets/fad0a49d-f0f0-4b0e-965d-6db2eccb9634) | ![image](https://github.com/user-attachments/assets/94515261-987c-4e0b-b24d-b65154f88f57)
:---|:---:|:---:
Pad speech: | _Base Drum 1 pad button_ | _Side Stick pad button_
Footer speech: | _Shortcut B; MIDI C2; Base Drum 1 footer button_ | _No shortcut; MIDI C-sharp 2; Side Stick footer button_

The screen reader might say row and column numbers afterwards, which is fine. You can test with your [built-in screen reader](https://github.com/musescore/MuseScore/wiki/Accessibility-testing#popular-screen-readers).

mathesoncalum commented:
Thanks for this @shoogle!  I'll be addressing some of these points in upcoming "refinements PRs"...

1. We'll address this after the merging of #22050 as this will change the navigation implementation slightly (see `contentNavigationPanelOrderStart` in that PR).

2. I'll address this in an upcoming refinements PR

3. I'm aware of the lost focus and I'll also be addressing this a refinements PR. As for the highlight behaviour, the spec suggests that the highlight _remains_ after confirming the swap:

![Screenshot 2024-12-05 at 12 48 41](https://github.com/user-attachments/assets/d61d1db9-b424-4dc6-a069-77116d9f8309)

4. This was disabled after #25608 - some design considerations need to be made here before re-implementing.

5. Looks good to me but the spec recommends something slightly different (namely including the row/column). Happy to include tweaks to this in a refinements PR.

![Screenshot 2024-12-05 at 12 49 54](https://github.com/user-attachments/assets/604efc96-317f-43bb-a7d6-23e65550b3d9)

I don't see any of these as blockers so I'm going to go ahead and merge - I'll coordinate privately on some of the points I've mentioned above. Cheers!
